### PR TITLE
Update PlatformServices.Desktop.Component.Tests to use new test fwk

### PR DESCRIPTION
**Pull request by Evangelink**
This PR also touches some other projects because these tests are referencing old versions of MSTest which is loaded in memory preventing the correct version (local) one to be loaded. This was ok up to now because we didn't do any big change (e.g. file rename).

### CODE REVIEW

File: TestFx.sln [hunk 1]
```diff
@@ -157,8 +163,6 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "DoNotParallelizeTestProject
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "DeploymentTestProject", "test\E2ETests\TestAssets\DeploymentTestProject\DeploymentTestProject.csproj", "{3FCE3987-7C8C-4E12-B54A-C6EC2D9FC7A6}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "CompatTestProject", "test\E2ETests\TestAssets\CompatTestProject\CompatTestProject.csproj", "{2D2C5B73-F1F1-47C8-BC5C-A172E9BB3D16}"
```
Comment (Evangelink) [inline TestFx.sln:158]: Removed from solution but kept locally so we avoid its content to be discovery and so loaded in memory.

File: test/ComponentTests/TestAssets/TestProjectForAssemblyResolution/TestProjectForAssemblyResolution.csproj [hunk 2]
```diff
@@ -7,9 +8,4 @@
     <AssemblyOriginatorKeyFile>$(RepoRoot)scripts\build\key.snk</AssemblyOriginatorKeyFile>
   </PropertyGroup>
 
-  <ItemGroup>
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="$(Legacy__MicrosoftNETTestSdkVersion)" />
```
Comment (Evangelink) [inline test/ComponentTests/TestAssets/TestProjectForAssemblyResolution/TestProjectForAssemblyResolution.csproj:11]: We actually don't care if that's a test project, all we care about is that the project type is loaded in memory because specified in the runsettings. Removing this reference avoid Test Explorer to discover and load the dlls in memory, causing side effects to our integration test.

File: test/ComponentTests/PlatformServices.Desktop.Component.Tests/App.config [hunk 3]
```diff
@@ -1,18 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
```
Comment (engyebrahim) [inline test/ComponentTests/PlatformServices.Desktop.Component.Tests/App.config:1]: please can you explain, why did you delete this file?

File: test/ComponentTests/PlatformServices.Desktop.Component.Tests/DesktopTestSourceHostTests.cs [hunk 4]
```diff
@@ -105,7 +96,7 @@ private static string GetTestAssemblyPath(string assemblyName)
         return Path.Combine(testAssetPath, assemblyName);
     }
 
-    private Mock<IRunSettings> GetMockedIRunSettings(string runSettingxml)
+    private static Mock<IRunSettings> GetMockedIRunSettings(string runSettingxml)
```
Comment (engyebrahim) [inline test/ComponentTests/PlatformServices.Desktop.Component.Tests/DesktopTestSourceHostTests.cs:99]: please can you explain, why we need it to be static?

File: test/ComponentTests/PlatformServices.Desktop.Component.Tests/DesktopTestSourceHostTests.cs [hunk 5]
```diff
@@ -75,26 +67,25 @@ public void ChildDomainResolutionPathsShouldHaveSearchDirectoriesSpecifiedInRuns
 
         var assemblyResolution = "ComponentTests\\TestProjectForAssemblyResolution.dll";
         var asm = Assembly.LoadFrom(assemblyResolution);
-        var type = asm.GetType("PlatformServices.Desktop.ComponentTests.TestProjectForAssemblyResolution");
+        var type = asm.GetType("PlatformServices.Desktop.ComponentTests.SerializableTypeThatShouldBeLoaded");
```
Comment (engyebrahim) [inline test/ComponentTests/PlatformServices.Desktop.Component.Tests/DesktopTestSourceHostTests.cs]: please can you explain, the difference between the two of them

File: test/ComponentTests/PlatformServices.Desktop.Component.Tests/ReflectionUtilityTests.cs [hunk 6]
```diff
@@ -294,33 +268,33 @@ private Assembly ReflectionOnlyOnResolve(object sender, ResolveEventArgs args)
         return null;
     }
 
-    private string[] GetAttributeValuePairs(object[] attributes)
+    private static string[] GetAttributeValuePairs(object[] attributes)
```
Comment (engyebrahim) [inline test/ComponentTests/PlatformServices.Desktop.Component.Tests/ReflectionUtilityTests.cs:271]: the same question why we need using static

File: test/ComponentTests/TestAssets/SampleFrameworkExtensions/Properties/AssemblyInfo.cs [hunk 7]
```diff
@@ -1,27 +0,0 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (engyebrahim) [inline test/ComponentTests/TestAssets/SampleFrameworkExtensions/Properties/AssemblyInfo.cs:1]: and please also explain, why we deleted this

**Review by engyebrahim — DISMISSED** [review]

File: test/ComponentTests/PlatformServices.Desktop.Component.Tests/App.config (hunk 3 above)
Comment (Evangelink) [reply test/ComponentTests/PlatformServices.Desktop.Component.Tests/App.config:1]: I did some cleaning, this file is actually no longer relevant (was already the case with main).

File: test/ComponentTests/PlatformServices.Desktop.Component.Tests/DesktopTestSourceHostTests.cs (hunk 4 above)
Comment (Evangelink) [reply test/ComponentTests/PlatformServices.Desktop.Component.Tests/DesktopTestSourceHostTests.cs:99]: Just took the liberty to address some code analyzers recommendation (method doesn't use any member so can be made static. `static` members are supposed to be faster so it's good to use them when possible).

File: test/ComponentTests/PlatformServices.Desktop.Component.Tests/ReflectionUtilityTests.cs (hunk 6 above)
Comment (Evangelink) [reply test/ComponentTests/PlatformServices.Desktop.Component.Tests/ReflectionUtilityTests.cs:271]: Just took the liberty to address some code analyzers recommendation (method doesn't use any member so can be made static. `static` members are supposed to be faster so it's good to use them when possible).

File: test/ComponentTests/TestAssets/SampleFrameworkExtensions/Properties/AssemblyInfo.cs (hunk 7 above)
Comment (Evangelink) [reply test/ComponentTests/TestAssets/SampleFrameworkExtensions/Properties/AssemblyInfo.cs:1]: This file is not required and wasn't bringing any benefits so I removed it to cleanup :)

File: test/ComponentTests/PlatformServices.Desktop.Component.Tests/DesktopTestSourceHostTests.cs (hunk 5 above)
Comment (Evangelink) [reply test/ComponentTests/PlatformServices.Desktop.Component.Tests/DesktopTestSourceHostTests.cs]: Definitely :) Basically we don't need the project we load to be a test (see https://github.com/microsoft/testfx/pull/1252/files#r970936904) so to make it more obvious I decided to rename the type so we don't think it's a test that is missing attributes or something.
